packages/grafana-runtime/src/utils/queryResponse.ts

```diff
@@ -93,6 +94,79 @@ export function toDataQueryResponse(res: any): DataQueryResponse {
   return rsp;
 }
 
+/**
+ * Parse the results from /api/ds/query into a DataQueryResponse ordering it by query input RefID's.
+ *
+ * @public
+ */
+export function toDataQueryResponseOrdered(queries: DataQuery[], res: any): DataQueryResponse {
+  const rsp: DataQueryResponse = { data: [], state: LoadingState.Done };
+  if (res.data?.results) {
```

Comment:
nit: Maybe do a quick exit here
```suggestion
  if (!res.data?.results) {
    return ...
  }
```

Reply:
Not sure what to return here if exit early. The not ok response could still be applicable if data contains results

Reply:
Hmm, but this is the case when there are no results. So you should just return `rsp`, I guess?

Reply:
I see now this is outdated, never mind.